chess_engine/main.js

```diff
@@ -0,0 +1,837 @@
+"use strict";
```

Comment:
It would be much easier to review code, when 3rd party code was used, and link to the source